sql/core/src/test/scala/org/apache/spark/sql/DataFrameSuite.scala

```diff
@@ -2439,6 +2439,17 @@ class DataFrameSuite extends QueryTest
     val nestedDecArray = Array(decSpark)
     checkAnswer(Seq(nestedDecArray).toDF(), Row(Array(wrapRefArray(decJava))))
   }
+
+  test("SPARK-31750: eliminate UpCast if child's dataType is DecimalType") {
+    withTempPath { f =>
+      sql("select cast(11111111111111111111111111111111111111 as decimal(38, 0)) as d")
```

Comment:
this test can still reproduce the bug even if we use `1` instead of `1111...`?

Reply:
Yes. It depends on the precision/scale rather than the value itself.

Reply:
I can make it shorter.